src/frontend/src/binder/query.rs

```diff
@@ -76,39 +78,54 @@ impl Binder {
             }),
             BoundSetExpr::Values(_) => {}
         };
+        let mut order_exprs = vec![];
+        let schema_len = body.schema().len();
         let order = query
             .order_by
             .into_iter()
-            .map(|order_by_expr| self.bind_order_by_expr(order_by_expr, &name_to_index))
+            .map(|order_by_expr| {
+                self.bind_order_by_expr(order_by_expr, &name_to_index, &mut order_exprs, schema_len)
+            })
             .collect::<Result<_>>()?;
         Ok(BoundQuery {
             body,
             order,
             limit,
             offset,
+            order_exprs,
         })
     }
 
     fn bind_order_by_expr(
         &mut self,
         order_by_expr: OrderByExpr,
         name_to_index: &HashMap<String, usize>,
+        order_exprs: &mut Vec<ExprImpl>,
+        schema_len: usize,
     ) -> Result<FieldOrder> {
         let direct = match order_by_expr.asc {
             None | Some(true) => Direction::Asc,
             Some(false) => Direction::Desc,
         };
-        let name = match order_by_expr.expr {
-            Expr::Identifier(name) => name.value,
+        let index = match order_by_expr.expr {
+            Expr::Identifier(name) => *name_to_index.get(&name.value).ok_or_else(|| {
+                ErrorCode::ItemNotFound(format!("Output column \"{}\"", name.value))
```

Comment:
This should also get a second chance to `extra_order_exprs.push(self.bind_expr(expr)?)`.

For example: `select x from t order by y;`